### PR TITLE
adds signup_resource.inc, SignupTransformer.php, and registers it in …

### PR DESCRIPTION
**Pull request by chloealee**
…dosomething_api.module
#### What does this PR do?

Creates the `signup.resource.inc` and `SignupTransformer.php` files. Registers this in `dosomething_api.module` 
#### How should this be manually tested?

`signup` should be it's own category highlighted in light blue in on the Services page: http://dev.dosomething.org:8888/admin/structure/services/list/drupalapi/resources. Dropdown should include `retrieve` and `index`.
#### What are the relevant tickets?

First step for #6110. Endpoint still needs to be enabled and parameters need to be commented in.

### CODE REVIEW

**Comment by weerd** [comment]
Remove the weird `._signup_resource.inc` hidden file that got committed.

File: lib/modules/dosomething/dosomething_api/resources/signup_resource.inc [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<?php
+
+function _signup_resource_definition() {
+  $signup_resource = [];
+  $signup_resource['signup'] = [
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_api/resources/signup_resource.inc]: make it plural! all endpoints are best to be kept plural as a general rule for REST APIs and standard to follow :)

**Comment by chloealee** [comment]
ah, thanks @weerd ! For some reason that always happens to me! Deleted it :)

**Comment by weerd** [comment]
Looks good! :+1: 

Didn't really look to much into the `SignupTransformer` stuff, cause it doesn't really matter just yet. Mostly just need the resource file and to update the module before we can activate the endpoint :)

**Pull request merged**